Avoid modifying LCC with Obj Func Incentives

adfarth opened the pull request:
- We previously had some of the objective function modifiers within the "Cost" equation and some were added on using `add_to_expression!( Objective,  ...` 
- Some, but not all, of these were then subtracted out from the LCC in `results/financial.jl`
- This PR reverts to how these modifiers were previously included to avoid modifying `Costs` (instead using `@objective(m, Min, m[:Costs] + m[:ObjectiveIncentives] )`)
- The only objective fn incentive that's still removed from the lcc in `results/financial.jl` is `0.0001 * value(m[Symbol("MinChargeAdder"*_n)])` since this is contained within the `TotalElecBill`

Bill-Becker reviewed (commented):
Looks good. A couple of things:

1. Did you run the test_with_xpress.jl locally? (Did they pass?)
2. Thoughts on the "Incentives" part of "ObjectiveIncentives" being confusing with actual incentives like tax/production incentives? Maybe there is a standard terminology in linear programming for this type of thing ("Bias", "Steer", "Influence"...?). If you can't find anything, I'm Ok with keeping "Incentives", but my first reaction was "why is incentives not included in the Costs?.

adfarth commented:
> Looks good. A couple of things:
> 
> 1. Did you run the test_with_xpress.jl locally? (Did they pass?)
> 2. Thoughts on the "Incentives" part of "ObjectiveIncentives" being confusing with actual incentives like tax/production incentives? Maybe there is a standard terminology in linear programming for this type of thing ("Bias", "Steer", "Influence"...?). If you can't find anything, I'm Ok with keeping "Incentives", but my first reaction was "why is incentives not included in the Costs?.

I have now run test_with_xpress.jl locally. One test failed due to the LCC changing with this PR. Fixed in https://github.com/NREL/REopt.jl/pull/345/commits/4271de024a847ebe4ebdcb93c677b4b57f2edd63.

I changed "ObjectiveIncentives" to "ObjectivePenalties"
